uint32 frame_rate
---
int32 result